### PR TITLE
Cleanup and corner case of RSA-PSS handling

### PR DESCRIPTION
**Pull request by Jakuje**
#### Description

This is a bit mixed bag, but I can split to smaller PRs if preferred.

Changes include:

* Throw away the ad-hoc selfsigning that was initially introduced for RSA-PSS certificates and due to certtool limitations. 
* Remove needless dependency on p11-kit-server
* Simplifies tests definitions
* RSA-PSS certs creation corner cases

There are still things that are not handled in ideal way:

 * The ALLOWED_MECHANISMS quirk (see top commit message)

#### Checklist

- [X] Code modified for feature
- [X] Test suite updated with functionality tests
- [X] Test suite updated with negative tests

#### Reviewer's checklist:

- [ ] ~Any issues marked for closing are addressed~
- [x] There is a test suite reasonably covering new functionality or modifications
- [ ] ~This feature/change has adequate documentation added~
- [x] Code conform to coding style that today cannot yet be enforced via the check style test
- [x] Commits have short titles and sensible commit messages
- [ ] Coverity Scan has run if needed (code PR) and no new defects were found

### CODE REVIEW

**Review by simo5 — CHANGES_REQUESTED** [review]
Mostly ok, one nit and one thing to drop, IMO

File: src/objects.c [hunk 1]
```diff
@@ -1083,7 +1083,8 @@ CK_RV p11prov_obj_from_handle(P11PROV_CTX *ctx, P11PROV_SESSION *session,
             if (ret == CKR_OK) {
                 obj->attrs[obj->numattrs] = a[0].attr;
                 obj->numattrs++;
-            } else if (ret == CKR_ATTRIBUTE_TYPE_INVALID) {
+            } else if (ret == CKR_ATTRIBUTE_TYPE_INVALID
+                       && obj->class == CKO_PRIVATE_KEY) {
```
[the remark this answers is not in the document]
Comment (Jakuje) [reply src/objects.c:1087]: Filled latchset/kryoptic#159 to handle this in kryoptic.

**Review by simo5 — APPROVED** [review]
LGTM
